File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 4/24/2026, 8:14:30 AM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ones = 0, maxOnes = 0;
        for (int item : nums) {
            if (item)
                ones++;
            else {
                if (ones > maxOnes)
                    maxOnes = ones;
                ones = 0;
            }
        }
        if (ones > maxOnes)
            maxOnes = ones;
        return maxOnes;
    }
};

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
// Last updated: 4/24/2026, 8:14:27 AM
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = Integer.MIN_VALUE;
        int count = 0;
        for (int i : nums) {
            if (i == 0)
                count = 0;
            else {
                count++;
                max = Math.max(max, count);
            }
        }
        return max < 0 ? 0 : max;
    }
}